src/gui/editors/SongEditor.cpp

```diff
@@ -293,6 +300,75 @@ void SongEditor::scrolled( int new_pos )
 
 
 
+void SongEditor::updateRubberband()
+{
+	if (rubberBandActive())
+	{
+		//take care of the zooming
+		int originX = m_origin.x();
+		if (m_currentZoomingValue != m_zoomingModel->value())
+		{
+			originX = m_trackHeadWidth + (originX - m_trackHeadWidth)
+					* m_zoomLevels[m_zoomingModel->value()] / m_zoomLevels[m_currentZoomingValue];
+		}
+
+		//take care of the scrollbar position
+		int hs = (m_leftRightScroll->value() - m_scrollPos.x()) * pixelsPerTact();
+		int vs = contentWidget()->verticalScrollBar()->value() - m_scrollPos.y();
+
+		//the adjusted origin point
+		QPoint origin = QPoint(qMax(originX - hs, m_trackHeadWidth), m_origin.y() - vs);
+
+		//paint the rubber band rect
+		rubberBand()->setGeometry(QRect( origin,
+										 contentWidget()->mapFromParent(QPoint(m_mousePos.x(), m_mousePos.y()))
+										).normalized());
+
+		//the index of the TrackView the mouse is hover
+		const TrackView * tv = trackViewAt(m_mousePos.y() - m_timeLine->height());
+		int rubberBandTrackview = 0;
+		if (tv)
+		{
+			for (auto it : trackViews())
+			{
+				if (it == tv) {break;}
+				rubberBandTrackview++;
+			}
+		}
+		else
+		{
+			rubberBandTrackview = (m_mousePos.y() < m_timeLine->height() ? 0 : trackViews().count());
+		}
+
+		//the miditime the mouse is hover
+		MidiTime rubberbandMidipos = MidiTime((m_mousePos.x() - m_trackHeadWidth) / pixelsPerTact()
+											  * MidiTime::ticksPerTact()) + m_currentPosition ;
+		//collect all Tcos
+		auto l = findChildren<selectableObject *>();
+
+		//are tcos in the rect of selection?
+		for (auto &it : l)
+		{
+			TrackContentObjectView * tco = dynamic_cast<TrackContentObjectView*>(it);
+
+			if (trackViews().indexOf(tco->getTrackView())     >= qMin(m_rubberBandStartTrackview, rubberBandTrackview)
```

Comment:
What's the meaning of this statement? `X >= Y` and then `X <= Y`? Am I missing something? 
Also, check for `nullptr` after `dynamic_cast`.

Reply:
You can make the selection from top to bottom or from bottom to top. The right side of the if-statement is the exact opposite in the second if-statement. Is it in the first statement `m_rubberBandStartTrackview` so it is mandatory in the second if-statement `rubberbandTrackView` and vice versa. So you can test whether the selection is made from below or from above.

But maybe I missed something? It gave me a real headache. Perhaps there is a simpler way?

Reply:
I don't understand: the condition `(X >= Y && X <= Y)` will be true only if `X==Y`.

Reply:
My bad; missed something.